lib/client/api.go

```diff
@@ -1940,6 +1954,43 @@ func (tc *TeleportClient) SSH(ctx context.Context, command []string, runLocally
 	return tc.runShell(ctx, nodeClient, types.SessionPeerMode, nil, nil)
 }
 
+func (tc *TeleportClient) shouldGuessCluster(ctx context.Context) bool {
+	if tc.ExplicitSiteName {
+		return false
+	}
+	pr, err := tc.Ping(ctx)
+	if err != nil {
+		return false
+	}
+	return pr.Auth.LoadAllHostCAs
+}
+
+// tryConnectToNodeAnyCluster attempts to connect to a node in an unknown cluster.
+func (tc *TeleportClient) tryConnectToNodeAnyCluster(ctx context.Context, proxyClient *ProxyClient, nodeAddr string) (*NodeClient, error) {
+	sites, err := proxyClient.GetSites(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var errors []error
+	for _, site := range sites {
+		// Assume that the current cluster was already checked.
+		if site.Name == tc.SiteName {
+			continue
+		}
+		tc.localAgent.UpdateCluster(site.Name)
+		nodeClient, err := proxyClient.ConnectToNode(
```

Comment:
That can take a while if there are many cluster clusters. 

Instead of doing it sequentially and iterating over sites can we just load all CAS if the `tc.SiteName` was not provided from https://github.com/gravitational/teleport/blob/master/tool/tsh/tsh.go#L2874 ?